src/lib/support/DefaultStorageKeyAllocator.h

```diff
@@ -265,6 +265,45 @@ class DefaultStorageKeyAllocator
     // Terms and Conditions Acceptance Key
     // Stores the terms and conditions acceptance including terms and conditions revision, TLV encoded
     static StorageKeyName TermsAndConditionsAcceptance() { return StorageKeyName::FromConst("g/tc"); }
+
+    // TLS Clusters Certs Keys
+
+    // Number of root certs stored in table for a given endpoint, across all fabrics.
+    static StorageKeyName TlsRootCertEndpointCountKey(EndpointId endpoint) { return StorageKeyName::Formatted("g/tlsr/e/%x", endpoint); }
```

Comment:
Do we want to have the cluster directly poke storage here, or do we want to have configurable providers that a cluster instance talks to (with a default impl that then maybe uses these keys)?

Because this setup here is not really conducive to a setup where anything that uses CHIPDeviceController implements the clusters that use these storage keys.  Is having that work a goal?

Reply:
Apologies not sure I understood; these keys are basically for `CertificateTableImpl`, and I would expect any cluster that wants to use this would use the `CertificateTable` instance to access, as was done with scenes (this is basically modeled after the scenes table 1:1)

Reply:
Scenes are pretty problematic as currently implemented too, unfortunately....  It's quite difficult to support that table thing along with CHIPDeviceController, and even worse if you have per-controller-segregated storage.

Now in practice things using CHIPDeviceController don't usually need to do scenes.  Will they need this cluster server?

Reply:
So theoretically this would only ever be done app-side, i.e. this is storage of the certs by the device, and as such shouldn't affect controller

Is that what you meant? I didn't understand "difficult to support that table thing along with CHIPDeviceController" - i.e. if something is a controller & an app, this should work, but in general storage to persistence would be done app-side

Reply:
The question is whether this would be implemented by something that currently uses controllers (e.g. Echo, or a HomePod, or something like that), or whether it would be something running a Server instance.  For example, such devices typically implement the server side of clusters like OTA Software Update Provider.  Would they be implementing this one?

Reply:
My expectation (at least for now with cameras) is that nothing that uses controllers would implement this (i.e. hubs like echo/homepod) - i.e. they would request the certs from cameras and request the camera to add a cert through the client management cluster, but they would be client-side, not server-side

However, nothing should be blocking them from using this if they had to, right? i.e. they'd just use `CertificateTableImpl` to persist the endpoints into local storage?

I think I may be missing your concern, may be quicker offline

Reply:
Also made `mBuffer` private so client can't access.